evm-template/runtime/src/configs/xcm_config.rs

```diff
@@ -370,3 +385,154 @@ parameter_types! {
+// How to convert from CurrencyId to Location
+pub struct CurrencyIdToLocation<AssetXConverter>(sp_std::marker::PhantomData<AssetXConverter>);
+impl<AssetXConverter> sp_runtime::traits::Convert<CurrencyId, Option<Location>>
+    for CurrencyIdToLocation<AssetXConverter>
+where
+    AssetXConverter: sp_runtime::traits::MaybeEquivalence<Location, AssetId>,
+{
+    fn convert(currency: CurrencyId) -> Option<Location> {
+        match currency {
+            CurrencyId::SelfReserve => {
+                let multi: Location = SelfReserve::get();
+                Some(multi)
+            }
+            CurrencyId::ForeignAsset(asset) => AssetXConverter::convert_back(&asset),
+            CurrencyId::Erc20 { contract_address } => {
+                let mut location = Erc20XcmBridgePalletLocation::get(); // TODO: if we do not have an alternative, we need to import this pallet from moonbeam
+                location
+                    .push_interior(Junction::AccountKey20 {
+                        key: contract_address.0,
+                        network: None,
+                    })
+                    .ok();
+                Some(location)
+            }
+        }
+    }
+}
+
+/// Wrapper type around `LocationToAccountId` to convert an `AccountId` to type `H160`.
+pub struct LocationToH160;
+impl ConvertLocation<H160> for LocationToH160 {
+    fn convert_location(location: &Location) -> Option<H160> {
+        <LocationToAccountId as ConvertLocation<AccountId>>::convert_location(location)
+            .map(Into::into)
+    }
+}
+
+parameter_types! {
+    // This is the relative view of erc20 assets.
+    // Identified by this prefix + AccountKey20(contractAddress)
+    // We use the RELATIVE multilocation
+    pub Erc20XcmBridgePalletLocation: Location = Location {
+        parents:0,
+        interior: [
+            PalletInstance(<Erc20XcmBridge as PalletInfoAccess>::index() as u8)
+        ].into()
+    };
+
+    // To be able to support almost all erc20 implementations,
+    // we provide a sufficiently high gas limit.
+    pub Erc20XcmBridgeTransferGasLimit: u64 = 800_000;
+}
+
+impl pallet_erc20_xcm_bridge::Config for Runtime {
+    type AccountIdConverter = LocationToH160;
+    type Erc20MultilocationPrefix = Erc20XcmBridgePalletLocation;
+    type Erc20TransferGasLimit = Erc20XcmBridgeTransferGasLimit;
+    type EvmRunner = pallet_evm::runner::stack::Runner<Self>;
+}
+
+impl orml_xtokens::Config for Runtime {
+    type AccountIdToLocation = AccountIdToLocation<AccountId>;
+    type Balance = Balance;
+    type BaseXcmWeight = BaseXcmWeight;
+    type CurrencyId = CurrencyId;
+    type CurrencyIdConvert = CurrencyIdToLocation<AsAssetType<AssetId, AssetType, AssetManager>>;
+    type LocationsFilter = Everything;
+    type MaxAssetsForTransfer = MaxAssetsForTransfer;
+    type MinXcmFee = ParachainMinFee;
+    type RateLimiter = ();
+    type RateLimiterId = ();
+    type ReserveProvider = AbsoluteAndRelativeReserve<SelfLocationAbsolute>;
```

Comment:
You will need to modify this reserve provider to include DOT from Asset Hub and Bridged Assets from Asset Hub otherwise xtokens will not be able to interact with those assets.